pkgs/applications/office/libreoffice/darwin/update-utils.nix

```diff
@@ -0,0 +1,54 @@
+{ lib }:
+let
+  # extractLatestVersionFromHtml :: String -> String
+  extractLatestVersionFromHtml = htmlString:
+    let
+      majorMinorPatchGroup = "([0-9]+\\.[0-9]+\\.[0-9]+)";
+      splittedVersions = builtins.split "href=\"${majorMinorPatchGroup}" htmlString;
+      stableVersions = builtins.concatLists
+        (builtins.filter (e: builtins.isList e)
+          splittedVersions);
+    in
+    if stableVersions == [ ]
+    then abort "Failed to extract versions from html."
+    else lib.last (builtins.sort builtins.lessThan stableVersions);
+
+  # getHtml :: String -> String
+  getHtml = url:
+    builtins.readFile (builtins.fetchurl url);
+
+  # getLatestStableVersion :: String
+  getLatestStableVersion =
+    extractLatestVersionFromHtml
+      (getHtml "https://download.documentfoundation.org/libreoffice/stable/");
+
+  # extractSha256FromHtml :: String -> String
+  extractSha256FromHtml = htmlString:
+    let
+      sha256 = (builtins.match ".*([0-9a-fA-F]{64}).*" htmlString);
```

Comment:
This could be very easily done with grep and be easy to debug.

Reply:
Same reasons as above of trying out a nix only approach.

In my opinion this is also easy to debug because there are tests and the nix-repl is also quite helpful to try out/manually test these functions.